core/drivers/clk/clk-stm32mp13.c

```diff
@@ -0,0 +1,1727 @@
+
+		fdt_err = clk_stm32_parse_oscillator_fdt(fdt, osc_node,
+							 osc_data->name, osci);
+		if (fdt_err < 0)
+			panic();
+	}
+
+	return 0;
+}
+
+#define RCC_PLL_NAME_SIZE 12
+
+static int clk_stm32_load_vco_config(const void *fdt, int subnode,
+				     struct stm32_pll_vco *vco)
+{
+	int ret = 0;
+
+	ret = _fdt_read_uint32_array(fdt, subnode, "divmn", vco->div_mn,
+				     (int)PLL_DIV_MN_NB);
+	if (ret != 0)
+		return ret;
+
+	ret = _fdt_read_uint32_array(fdt, subnode, "csg", vco->csg,
+				     (int)PLL_CSG_NB);
+
+	vco->csg_enabled = (ret == 0);
+
+	if (ret == -FDT_ERR_NOTFOUND)
+		ret = 0;
+
+	if (ret != 0)
+		return ret;
+
+	vco->status = RCC_PLLNCR_DIVPEN | RCC_PLLNCR_DIVQEN |
+		      RCC_PLLNCR_DIVREN | RCC_PLLNCR_PLLON;
+
+	vco->frac = _fdt_read_uint32_default(fdt, subnode, "frac", 0);
+
+	vco->src = _fdt_read_uint32_default(fdt, subnode, "src", UINT32_MAX);
+
+	return 0;
+}
+
+static int clk_stm32_load_output_config(const void *fdt, int subnode,
+					struct stm32_pll_output *output)
+{
+	return _fdt_read_uint32_array(fdt, subnode, "st,pll_div_pqr",
+				      output->output, (int)PLL_DIV_PQR_NB);
+}
+
+static int clk_stm32_parse_pll_fdt(const void *fdt, int subnode,
+				   struct stm32_pll_dt_cfg *pll)
+{
+	const fdt32_t *cuint = NULL;
+	int subnode_pll = 0;
+	int subnode_vco = 0;
+	int err = 0;
+
+	cuint = fdt_getprop(fdt, subnode, "st,pll", NULL);
+	if (!cuint)
+		return -FDT_ERR_NOTFOUND;
+
+	subnode_pll = fdt_node_offset_by_phandle(fdt, fdt32_to_cpu(*cuint));
+	if (subnode_pll < 0)
+		return -FDT_ERR_NOTFOUND;
+
+	cuint = fdt_getprop(fdt, subnode_pll, "st,pll_vco", NULL);
+	if (!cuint)
+		return -FDT_ERR_NOTFOUND;
+
+	subnode_vco = fdt_node_offset_by_phandle(fdt, fdt32_to_cpu(*cuint));
+	if (subnode_vco < 0)
+		return -FDT_ERR_NOTFOUND;
+
+	err = clk_stm32_load_vco_config(fdt, subnode_vco, &pll->vco);
+	if (err != 0)
+		return err;
+
+	err = clk_stm32_load_output_config(fdt, subnode_pll, &pll->output);
+	if (err != 0)
+		return err;
+
+	return 0;
+}
+
+static int stm32_clk_parse_fdt_all_pll(const void *fdt, int node,
+				       struct stm32_clk_platdata *pdata)
+{
+	size_t i = 0;
+
+	for (i = PLL1_ID; i < pdata->npll; i++) {
+		struct stm32_pll_dt_cfg *pll = pdata->pll + i;
+		char name[RCC_PLL_NAME_SIZE];
+		int subnode = 0;
+		int err = 0;
+
+		snprintf(name, sizeof(name), "st,pll@%d", i);
+
+		subnode = fdt_subnode_offset(fdt, node, name);
+		if (!_fdt_check_node(fdt, subnode))
+			continue;
+
+		err = clk_stm32_parse_pll_fdt(fdt, subnode, pll);
+		if (err != 0)
+			panic();
+	}
+
+	return 0;
+}
+
+static int stm32_clk_parse_fdt_opp(const void *fdt, int node,
+				   const char *opp_name,
+				   struct stm32_clk_opp_cfg *opp_cfg)
+{
+	int subnode = 0;
+	int nb_opp = 0;
+
+	node = fdt_subnode_offset(fdt, node, opp_name);
+	if (!_fdt_check_node(fdt, node))
+		return 0;
+
+	fdt_for_each_subnode(subnode, fdt, node) {
+		if (nb_opp >= MAX_OPP) {
+			EMSG("%d MAX opp in %s !", MAX_OPP, opp_name);
```

Comment:
remove ` !` from trace message. Avoid `!` in trace messages.

add `return -1;` or `panic();`